pandora/models/pandora.py

```diff
@@ -223,11 +226,26 @@ class AdItem(PlaylistModel):
     def is_ad(self):
         return True
 
-    def register_ad(self, station_id):
-        self._api_client.register_ad(station_id, self.tracking_tokens)
+    def register_ad(self, station_id=None):
+        if not station_id:
+            station_id = self.station_id
+        if self.tracking_tokens:
+            self._api_client.register_ad(station_id, self.tracking_tokens)
+        else:
+            raise ParameterMissing('No ad tracking tokens provided for '
+                                   'registration.')
 
     def prepare_playback(self):
-        self.register_ad(self.station_id)
```

Comment:
If `register_ad` doesn't raise an exception and doesn't make a server-call when it has no ad tracking you should be able to just call `register_ad` here without any error handling.